olp-cpp-sdk-core/tests/cache/DefaultCacheTest.cpp

```diff
@@ -263,40 +265,199 @@ TEST(DefaultCacheTest, MemSizeTest) {
+
+    // Removes "key4"
+    ASSERT_FALSE(cache.Get("key4", decoder).empty());
+    cache.Remove("key4");
+    ASSERT_TRUE(cache.Get("key4", decoder).empty());
+
+    // Remove nothing
+    ASSERT_FALSE(cache.Get("key2", decoder).empty());
+    ASSERT_FALSE(cache.Get("key3", decoder).empty());
+    ASSERT_FALSE(cache.Get("key5", decoder).empty());
+    ASSERT_FALSE(cache.Get("key7", decoder).empty());
+    cache.Remove("doesnotexist");
+    ASSERT_FALSE(cache.Get("key2", decoder).empty());
+    ASSERT_FALSE(cache.Get("key3", decoder).empty());
+    ASSERT_FALSE(cache.Get("key5", decoder).empty());
+    ASSERT_FALSE(cache.Get("key7", decoder).empty());
+
+    // Remove all
+    ASSERT_TRUE(cache.Clear());
+    ASSERT_TRUE(cache.Get("key2", decoder).empty());
+    ASSERT_TRUE(cache.Get("key3", decoder).empty());
+    ASSERT_TRUE(cache.Get("key5", decoder).empty());
+    ASSERT_TRUE(cache.Get("key7", decoder).empty());
+  }
+
+  {
+    SCOPED_TRACE("With protection");
+
+    fill_data();
+
+    // Protect key2, key3 and key1 (covering also key10)
+    ASSERT_TRUE(cache.Protect({"key2", "key3", "key1"}));
+
+    // Try remove "key1" && "key10"
+    ASSERT_FALSE(cache.Get("key1", decoder).empty());
+    ASSERT_FALSE(cache.Get("key10", decoder).empty());
+
+    cache.Remove("key1");
+    ASSERT_FALSE(cache.Get("key10", decoder).empty());
+    ASSERT_FALSE(cache.Get("key1", decoder).empty());
+
+    cache.Remove("key10");
+    ASSERT_FALSE(cache.Get("key10", decoder).empty());
+    ASSERT_FALSE(cache.Get("key1", decoder).empty());
+
+    // Removes "key4"
+    ASSERT_FALSE(cache.Get("key4", decoder).empty());
+    cache.Remove("key4");
+    ASSERT_TRUE(cache.Get("key4", decoder).empty());
+
+    // Try remove "key2" && "key3"
+    ASSERT_FALSE(cache.Get("key2", decoder).empty());
+    ASSERT_FALSE(cache.Get("key3", decoder).empty());
+
+    cache.Remove("key2");
+    cache.Remove("key3");
+
+    ASSERT_FALSE(cache.Get("key2", decoder).empty());
+    ASSERT_FALSE(cache.Get("key3", decoder).empty());
+
+    // Remove nothing
+    ASSERT_FALSE(cache.Get("key2", decoder).empty());
+    ASSERT_FALSE(cache.Get("key3", decoder).empty());
+    ASSERT_FALSE(cache.Get("key5", decoder).empty());
+    ASSERT_FALSE(cache.Get("key7", decoder).empty());
+    cache.Remove("doesnotexist");
+    ASSERT_FALSE(cache.Get("key2", decoder).empty());
+    ASSERT_FALSE(cache.Get("key3", decoder).empty());
+    ASSERT_FALSE(cache.Get("key5", decoder).empty());
+    ASSERT_FALSE(cache.Get("key7", decoder).empty());
+
+    // Remove all
+    ASSERT_TRUE(cache.Clear());
+    ASSERT_TRUE(cache.Get("key2", decoder).empty());
+    ASSERT_TRUE(cache.Get("key3", decoder).empty());
+    ASSERT_TRUE(cache.Get("key5", decoder).empty());
+    ASSERT_TRUE(cache.Get("key7", decoder).empty());
+  }
+}
+
 TEST(DefaultCacheTest, RemoveWithPrefix) {
-  olp::cache::DefaultCache cache;
+  olp::cache::CacheSettings settings;
+  settings.max_memory_cache_size = 0;
```

Comment:
Why disable memory cache? It would be good to know protected keys are not removed from InMemoryCache too.

Reply:
But there is no way to check if the data comes from the in-memory or from the mutable cache. Of course, it would make sense to only use in-memory and no mutable cache and vice-versa to guarantee that both ways it works. I can add another test to validate that the in-memory allone also works.